### PR TITLE
Added generic many to many field

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: example/apps/app/tests/__init__.py [hunk 1]
```diff
@@ -201,3 +204,107 @@ def test_generic_m2m_manager_should_get_object_or_none_accroding_to_model_class(
...
+        assert_equal(m2m_inst.related_objects.related_object1, related_object_inst1)
+
+        m2m_inst.related_objects.add(related_object2=related_object_inst2, related_object3=related_object_inst3)
+        assert_equal(m2m_inst.related_objects.count(), 3)
+        assert_equal(m2m_inst.related_objects.related_object1, related_object_inst1)
+        assert_equal(m2m_inst.related_objects.related_object2, related_object_inst2)
+        assert_equal(m2m_inst.related_objects.related_object3, related_object_inst3)
+
+        # Duplicate named objects are updated
+        m2m_inst.related_objects.add(related_object1=related_object_inst2)
```
Comment (Formulka) [inline example/apps/app/tests/__init__.py:226]: I have to remember this, it's convenient but a bit unexpected.

File: generic_m2m_field/models.py [hunk 2]
```diff
@@ -82,7 +100,13 @@ def _filter_or_exclude(self, negate, *args, **kwargs):
         return super()._filter_or_exclude(negate, *args, **kwargs)
 
 
-class GenericManyToManyManager(models.Manager):
+class BaseGenericManager(models.Manager):
+
+    def _is_related_manger(self):
```
Comment (Formulka) [inline generic_m2m_field/models.py]: `_is_related_manager`

File: generic_m2m_field/models.py [hunk 3]
```diff
@@ -92,8 +116,30 @@ def __init__(self, *args, **kwargs):
...
             self.remove = MethodType(remove_objs, self)
 
-    def _is_related_manger(self):
-        return self.__class__.__module__ == 'django.db.models.fields.related_descriptors'
+
+class NamedGenericManyToManyManager(BaseGenericManager):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self._is_related_manger():
```
Comment (Formulka) [inline generic_m2m_field/models.py]: `_is_related_manager`

File: example/apps/app/tests/__init__.py (hunk 1 above)
Comment (rubickcz) [reply example/apps/app/tests/__init__.py:226]: It makes sense for me, what should be expected behavior in your opinion?

**Review by rubickcz — APPROVED** [review]

File: example/apps/app/tests/__init__.py (hunk 1 above)
Comment (Formulka) [reply example/apps/app/tests/__init__.py:226]: I would just expect a separate method for updating, this is more convenient, though

**Review by Formulka — APPROVED** [review]

File: example/apps/app/tests/__init__.py (hunk 1 above)
Comment (matllubos) [reply example/apps/app/tests/__init__.py:226]: For me this is better than separate function. The second reason is that I need to override default django behavior. Without it exception will be returned and this is what we don't wan.

**Pull request merged**